test/integration/scheduler/scheduler_test.go

```diff
@@ -638,3 +638,177 @@ func TestNodeEvents(t *testing.T) {
+	}
+
+	// 4.1 create i1
+	i1, err := testutils.CreatePausePod(testCtx.ClientSet, testutils.InitPausePod(&testutils.PausePodConfig{
+		Name:      "i1",
+		Namespace: testCtx.NS.Name,
+		InitContainerPorts: []v1.ContainerPort{
+			{
+				ContainerPort: 8001,
+				HostPort:      8001,
+				Protocol:      v1.ProtocolTCP,
+			},
+		},
+	}))
+	if err != nil {
+		t.Fatalf("Failed to create Pod i1: %v", err)
+	}
+	// 4.2 verify i1 is unscheduable
+	if err := testutils.WaitForPodUnschedulable(testCtx.Ctx, testCtx.ClientSet, i1); err != nil {
+		t.Errorf("Pod %v got scheduled: %v", i1.Name, err)
+	}
+
+	// 5.1 delete c2
+	if err := testutils.DeletePod(testCtx.ClientSet, c2.Name, testCtx.NS.Name); err != nil {
+		t.Errorf("Pod %v did not delete: %v", c2.Name, err)
+	}
+	// 5.2 verify i1 is scheduled
+	err = testutils.WaitForPodToScheduleWithTimeout(testCtx.Ctx, testCtx.ClientSet, i1, time.Second*5)
+	if err != nil {
+		t.Errorf("Pod %s didn't schedule: %v", i1.Name, err)
+	}
+
+	// 6.1 create i2
+	i2, err := testutils.CreatePausePod(testCtx.ClientSet, testutils.InitPausePod(&testutils.PausePodConfig{
+		Name:      "i2",
+		Namespace: testCtx.NS.Name,
+		InitContainerPorts: []v1.ContainerPort{
+			{
+				ContainerPort: 8001,
+				HostPort:      8001,
+				Protocol:      v1.ProtocolTCP,
+			},
+		},
+	}))
+	if err != nil {
+		t.Fatalf("Failed to create Pod i2: %v", err)
+	}
+	// 6.2 verify i2 is unscheduable
+	if err := testutils.WaitForPodUnschedulable(testCtx.Ctx, testCtx.ClientSet, i2); err != nil {
+		t.Errorf("Pod %v got scheduled: %v", i2.Name, err)
+	}
+
+	// 7.1 delete i1
+	if err := testutils.DeletePod(testCtx.ClientSet, i1.Name, testCtx.NS.Name); err != nil {
+		t.Errorf("Pod %v did not delete: %v", i1.Name, err)
+	}
+	// 7.2 verify i2 is scheduled
+	err = testutils.WaitForPodToScheduleWithTimeout(testCtx.Ctx, testCtx.ClientSet, i2, time.Second*5)
+	if err != nil {
+		t.Errorf("Pod %s didn't schedule: %v", i2.Name, err)
+	}
+
+	// 8.1 create c3
+	c3, err := testutils.CreatePausePod(testCtx.ClientSet, testutils.InitPausePod(&testutils.PausePodConfig{
+		Name:      "c3",
+		Namespace: testCtx.NS.Name,
+		InitContainerPorts: []v1.ContainerPort{
+			{
+				ContainerPort: 8001,
+				HostPort:      8001,
+				Protocol:      v1.ProtocolTCP,
+			},
+		},
+	}))
+	if err != nil {
+		t.Fatalf("Failed to create Pod c3: %v", err)
+	}
+	// 8.2 verify c3 is unscheduable
+	if err := testutils.WaitForPodUnschedulable(testCtx.Ctx, testCtx.ClientSet, c3); err != nil {
+		t.Errorf("Pod %v got scheduled: %v", c3.Name, err)
+	}
+
+	// 9.1 delete i2
+	if err := testutils.DeletePod(testCtx.ClientSet, i2.Name, testCtx.NS.Name); err != nil {
+		t.Errorf("Pod %v did not delete: %v", i2.Name, err)
+	}
+	// 9.2 verify c3 is scheduled
+	err = testutils.WaitForPodToScheduleWithTimeout(testCtx.Ctx, testCtx.ClientSet, c3, time.Second*5)
+	if err != nil {
+		t.Errorf("Pod %s didn't schedule: %v", c3.Name, err)
+	}
+
+	// Clean up
```

Comment:
nit: Since we're not running multiple test cases using table-driven testing, cleanup might not be necessary here.

Reply:
Removed